Add explanation on fx:root (#35)
se-edu/guides: tutorials/javaFxPart4.md

   ![Controller for MainWindow](images/javafx/MainWindowController.png)

1. Let’s repeat the process for `DialogBox`.

>    The main difference here is that DialogBox checks `Use fx:root construct` and _does not define a controller class_. Checking this enables the use of the `fx:root` element, allowing you to reference a root element predefined by calling the `setRoot()` method. The controller can be programatically set using the `setController()` method. 
>    
>    More about `fx:root` on the documentation [Introduction to FXML | JavaFX 2.2](https://docs.oracle.com/javafx/2/api/javafx/fxml/doc-files/introduction_to_fxml.html#root_elements).

Reviewer: ```suggestion
   The main difference here is that we tick the checkbox `Use fx:root construct` and _do not define a controller class_. Ticking this enables the use of the `fx:root` element, allowing you to reference a root element predefined by calling the `setRoot()` method. The controller can be programatically set using the `setController()` method. 
   
   More about `fx:root` on the documentation [Introduction to FXML | JavaFX 2.2](https://docs.oracle.com/javafx/2/api/javafx/fxml/doc-files/introduction_to_fxml.html#root_elements).
```
Reason: `checks` can be interpreted as _verifies_.

Reviewer: Can/should we also explain why we do the DialogBox differently? _Must_ we do it this way for DialogBox or we are simply illustrating a different technique?

Reviewer: > ... by calling the `setRoot()` method.

It's not clear if you mean _this allows you to call the `setRoot()` method_ or _this has the same effect as calling the `setRoot()` method_.

Author: > Can/should we also explain why we do the DialogBox differently? _Must_ we do it this way for DialogBox or we are simply illustrating a different technique?

This is done differently since DialogBox is a custom component we build that will be reused.

Author: > > ... by calling the `setRoot()` method.
> 
> It's not clear if you mean _this allows you to call the `setRoot()` method_ or _this has the same effect as calling the `setRoot()` method_.

Use of the `fx:root` element allows us to reference a root element. This root element will be predefined by calling the `setRoot()` method before we `load()`.
